modules/node_groups/variables.tf

```diff
@@ -34,3 +34,11 @@ variable "node_groups" {
   type        = any
   default     = {}
 }
+
+# Hack for a homemade `depends_on` https://discuss.hashicorp.com/t/tips-howto-implement-module-depends-on-emulation/2305/2
+# Will be removed when Terraform will add support of module's `depends_on`https://github.com/hashicorp/terraform/issues/10462
```

Comment:
The module cannot depend on the aws_auth. Output from the module is used to generate the configmap's roles key.

Reply:
fixed

Reply:
Humm after reading some issues, I think your comment https://github.com/terraform-aws-modules/terraform-aws-eks/issues/852#issuecomment-621076125 still apply. Terraform won't go into dependency cycle because the [`aws_auth_roles` output from MNG](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/node_groups/outputs.tf#L6-L14) is built with locals.